neurots/generate/tree.py

```diff
@@ -248,8 +248,19 @@ def next_point(self):
         for section_grower in ordered_sections:
             # the current section_grower is generated
             # In here the stop criterion can be modified accordingly
+
+            if section_grower.process == "major" and "cut_major_length" in self.params:  # pragma: no cover
+                # this makes an early termination of major branch
+                # it needs to revert the value after getting the state to preserve
+                # the original topology
+                _term = section_grower.stop_criteria["TMD"].term
+                section_grower.stop_criteria["TMD"].term = self.params["cut_major_length"]
+
             state = self.growth_algo.extend(section_grower)
 
+            if section_grower.process == "major" and "cut_major_length" in self.params:
+                section_grower.stop_criteria["TMD"].term = _term
```

Comment:
```suggestion
            if section_grower.process == "major" and "cut_major_length" in self.params:  # pragma: no cover
                # this makes an early termination of major branch
                # it needs to revert the value after getting the state to preserve
                # the original topology
                _term = section_grower.stop_criteria["TMD"].term
                section_grower.stop_criteria["TMD"].term = self.params["cut_major_length"]
            else:
                _term = None

            state = self.growth_algo.extend(section_grower)

            if _term is not None:  # pragma: no cover
                section_grower.stop_criteria["TMD"].term = _term
```
Another idea to improve perfs a bit (avoid duplicated checks)